src/dpp/message.cpp

```diff
@@ -725,10 +725,10 @@ attachment::attachment(struct message* o, json *j) : attachment(o) {
 
 void attachment::download(http_completion_event callback) const {
 	/* Download attachment if there is one attached to this object */
-	if (!owner->owner) {
+	if (owner == nullptr || owner->owner == nullptr) {
 		throw dpp::logic_exception("attachment has no owning message/cluster");
 	}
-	if (callback && this->id && !this->url.empty()) {
+	if (callback && !this->url.empty()) {
```

Comment:
idk for what this was o:

Reply:
it checks there's an id for the attachment, you can't download without one. the check needs to remain...

Reply:
okay. its now back in